python/revng/daemon/schema.graphql

```diff
@@ -19,8 +25,30 @@ type Target {
 type Mutation {
     uploadB64(input: String!, container: String!): Boolean!
     uploadFile(file: Upload, container: String!): Boolean!
-    runAnalysis(step: String!, analysis: String!, containerToTargets: String, options: String): String!
-    runAnalysesList(name: String!, options: String): String!
+    runAnalysis(step: String!, analysis: String!, containerToTargets: String, options: String): AnalysisResult!
+    runAnalysesList(name: String!, options: String): AnalysisResult!
+}
+
+union AnalysisResult = Diff | SimpleError | DocumentError
```

Comment:
What's the ratio for handling errors in-band?

GraphQL seems to have [its own error reporting](https://learning.postman.com/open-technologies/blog/graphql-error-handling/). Have you considered using it?

Reply:
Yes, however it does not provide strong typing for the errors at the schema level, so it would have been considerably worse. The current implementation:
* Makes it clear which operations are failable right on the schema
* Makes it clear what the structure of the errors is, providing all the GraphQL goodies such as aliases etc.

Reply:
OK, add a comment about this.

Reply:
Done